autogen/agentchat/contrib/capabilities/context_handling.py

```diff
@@ -0,0 +1,112 @@
+import sys
+from termcolor import colored
+from typing import Dict, Optional, Union, List, Tuple, Any
+from autogen.agentchat.contrib.capabilities.agent_capability import AgentCapability
+from autogen import ConversableAgent
+from autogen import token_count_utils
+
+
+class TransformChatHistory:
+    """
+    An agent's chat history with other agents is a common context that it uses to generate a reply.
+    This capability allows the agent to transform its chat history prior to using it to generate a reply.
+    It does not permanently modify the chat history, but rather processes it on every invocation.
+
+    """
+
+    def __init__(
+        self, max_tokens_per_message: int = sys.maxsize, max_messages: int = sys.maxsize, max_tokens: int = sys.maxsize
```

Comment:
I think if the default is `sys.maxsize` then it may as well use `Optional[int] = None` to indicate we are not enforcing the constraint if the argument is not provided.

Reply:
Another thing I learned from @davorrunje is that if we anticipate the argument list to change in the future, we can add `*` before the keyword arguments to consume all positional arguments and make sure people are using keyword arguments rather than positional arguments.